### PR TITLE
RevEng: Updating command-line table-selection syntax

### PR DESCRIPTION
**Pull request by lajones**
As per the design meeting - see https://github.com/aspnet/EntityFramework/wiki/Design-Meeting-Notes-(October-1,-2015)#table-selection-in-reverse-engineering

Fix for #3296. Also provides fix for #3227.

### CODE REVIEW

File: src/EntityFramework.Relational.Design/ReverseEngineering/Internal/TableSelectionSet.cs [hunk 1]
```diff
@@ -9,98 +9,16 @@ namespace Microsoft.Data.Entity.Relational.Design.ReverseEngineering.Internal
...
-            new List <TableSelection> { TableSelection.InclusiveAll };
-        private static readonly List<TableSelection> _exclusiveAll =
-            new List<TableSelection> { TableSelection.ExclusiveAll };
-
-        private List<TableSelection> _inclusiveSelections = new List<TableSelection>();
-        private List<TableSelection> _exclusiveSelections = new List<TableSelection>();
-
-        public virtual void AddSelections([CanBeNull] params TableSelection[] tableSelections)
+        public TableSelectionSet(
+            [CanBeNull] List<string> schemas, [CanBeNull] List<string> tables)
```
Comment (natemcmaster) [inline src/EntityFramework.Relational.Design/ReverseEngineering/Internal/TableSelectionSet.cs]: Can we just create a default, parameterless constructor?

File: test/EntityFramework.Relational.Design.Tests/ReverseEngineering/Internal/TableSelectionSetTests.cs [hunk 2]
```diff
@@ -9,340 +8,11 @@ namespace Microsoft.Data.Entity.Relational.Design.ReverseEngineering.Internal.Te
...
-            Assert.False(tableSelectionSet.Allows("schema2", "table3"));
-            Assert.False(tableSelectionSet.Allows("schema3", "table1"));
-            Assert.False(tableSelectionSet.Allows("schema3", "table2"));
-            Assert.False(tableSelectionSet.Allows("schema3", "table3"));
-            Assert.False(tableSelectionSet.Allows("schemaN", "table1"));
-            Assert.False(tableSelectionSet.Allows("schemaN", "table2"));
-            Assert.False(tableSelectionSet.Allows("schemaN", "table3"));
+            var tableSelectionSet = new TableSelectionSet(null, null);
+            Assert.Equal(0, tableSelectionSet.Schemas.Count);
+            Assert.Equal(0, tableSelectionSet.Tables.Count);
```
Comment (natemcmaster) [inline test/EntityFramework.Relational.Design.Tests/ReverseEngineering/Internal/TableSelectionSetTests.cs]: Trivial test. We can just delete altogether.

**Comment by natemcmaster** [comment]
This is looking good @lajones

File: src/EntityFramework.Relational.Design/ReverseEngineering/Internal/TableSelectionSet.cs (hunk 1 above)
Comment (lajones) [inline src/EntityFramework.Relational.Design/ReverseEngineering/Internal/TableSelectionSet.cs]: Thought about it - but I'd rather people used static InclusiveAll object for that.

Comment (natemcmaster) [inline src/EntityFramework.Relational.Design/ReverseEngineering/Internal/TableSelectionSet.cs]: This is just wrapping `null`. We can safely assume a null reference for TableSelectionSet = include all.

File: test/EntityFramework.Relational.Design.Tests/ReverseEngineering/Internal/TableSelectionSetTests.cs (hunk 2 above)
Comment (lajones) [inline test/EntityFramework.Relational.Design.Tests/ReverseEngineering/Internal/TableSelectionSetTests.cs]: OK. I thought it wasn't obvious that a null would become an empty list. But I'm fine without it.

**Comment by natemcmaster** [comment]
:shipit:

**Comment by lajones** [comment]
Checked in with commit 1c2916b.

**Pull request closed without merging**